tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/virtual/VirtualThreadServlet.java

```diff
@@ -0,0 +1,468 @@
+
+@ManagedScheduledExecutorDefinition(name = "java:app/concurrent/ManagedScheduledExecutorAnnoPlatform", virtual = false)
+@ManagedScheduledExecutorDefinition(name = "java:app/concurrent/ManagedScheduledExecutorAnnoVirtual", virtual = true)
+
+@ManagedThreadFactoryDefinition(name = "java:app/concurrent/ThreadFactoryAnnoPlatform", virtual = false)
+@ManagedThreadFactoryDefinition(name = "java:app/concurrent/ThreadFactoryAnnoVirtual", virtual = true)
+
+@WebServlet("/VirtualThreadServlet")
+public class VirtualThreadServlet extends TestServlet {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Runnable NOOP_RUNNABLE = () -> {
+    };
+        
+    private static final int VERSION = Runtime.version().feature();
+
+    public void testPlatformExecutor() throws Exception {
+        ManagedExecutorService platformManagedExecutorAnno = InitialContext
+                .doLookup("java:app/concurrent/ManagedExecutorAnnoPlatform");
+        ManagedExecutorService platformManagedExecutorDD = InitialContext
+                .doLookup("java:app/concurrent/ManagedExecutorDDPlatform");
+        
+        assertNotNull(platformManagedExecutorAnno);
+        assertNotNull(platformManagedExecutorDD);
+
+        Future<Thread> future;
+
+        future = platformManagedExecutorAnno.submit(Thread::currentThread);
+        assertFalse(isVirtual(future.get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS)));
+
+        future = platformManagedExecutorDD.submit(Thread::currentThread);
+        assertFalse(isVirtual(future.get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS)));
+    }
+
+    public void testVirtualExecutor() throws Exception {
+        ManagedExecutorService virtualManagedExecutorAnno = InitialContext
+                .doLookup("java:app/concurrent/ManagedExecutorAnnoVirtual");
+        ManagedExecutorService virtualManagedExecutorDD = InitialContext
+                .doLookup("java:app/concurrent/ManagedExecutorDDVirtual");
+        
+        assertNotNull(virtualManagedExecutorAnno);
+        assertNotNull(virtualManagedExecutorDD);
+        
+        Thread annoThread = virtualManagedExecutorAnno.submit(Thread::currentThread)
+                .get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        Thread ddThread = virtualManagedExecutorDD.submit(Thread::currentThread)
+                .get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        
+        if (VERSION == 17) { //TODO remove when Concurrency API supports only 21+
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(annoThread), "Should be impossible to get a virtual thread on Java 17");
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(ddThread), "Should be impossible to get a virtual thread on Java 17");
+            return;
+        }
+
+        // Java 21+
+        assertEquals(isVirtual(annoThread), isVirtual(ddThread));
```

Comment:
I don't think the specification makes any requirement that if an implementation decides to honor virtual=true in one place that it must also honor it in another place.  That said, it seems unlikely that an implementation would not comply with this given they are basically the same usage here.